<h1> Testing python </h1>

In [1]:
print("Hello World!")

Hello World!


In [2]:
pip install pandas

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 24.2 -> 25.0
[notice] To update, run: python.exe -m pip install --upgrade pip


In [3]:
pip install numpy

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 24.2 -> 25.0
[notice] To update, run: python.exe -m pip install --upgrade pip


In [4]:
pip install matplotlib

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 24.2 -> 25.0
[notice] To update, run: python.exe -m pip install --upgrade pip


In [5]:
pip install seaborn

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 24.2 -> 25.0
[notice] To update, run: python.exe -m pip install --upgrade pip


In [6]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [7]:
birds = pd.read_csv('datasets\\penguins.csv')

In [8]:
birds

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen            39.1           18.7   
1        2     Adelie  Torgersen            39.5           17.4   
2        3     Adelie  Torgersen            40.3           18.0   
3        4     Adelie  Torgersen             NaN            NaN   
4        5     Adelie  Torgersen            36.7           19.3   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream            55.8           19.8   
340    341  Chinstrap      Dream            43.5           18.1   
341    342  Chinstrap      Dream            49.6           18.2   
342    343  Chinstrap      Dream            50.8           19.0   
343    344  Chinstrap      Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  
0                181.0       3750.0    male  2007  
1                186.0       3800.0  female  2007  
2                195.0       3250.0  female  2007  
3                  NaN          NaN     NaN  2007  
4                193.0       3450.0  female  2007  
..                 ...          ...     ...   ...  
339              207.0       4000.0    male  2009  
340              202.0       3400.0  female  2009  
341              193.0       3775.0    male  2009  
342              210.0       4100.0    male  2009  
343              198.0       3775.0  female  2009  

[344 rows x 9 columns]

<h1> FILTERING DATA BASED FROM COLUMNS </h1>

In [9]:
birds['body_mass_g']

0      3750.0
1      3800.0
2      3250.0
3         NaN
4      3450.0
        ...  
339    4000.0
340    3400.0
341    3775.0
342    4100.0
343    3775.0
Name: body_mass_g, Length: 344, dtype: float64

<h1> CREATE OWN COLUMNS </h1>

In [10]:
birds['weight'] = birds['body_mass_g'] / 1000

In [11]:
birds

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen            39.1           18.7   
1        2     Adelie  Torgersen            39.5           17.4   
2        3     Adelie  Torgersen            40.3           18.0   
3        4     Adelie  Torgersen             NaN            NaN   
4        5     Adelie  Torgersen            36.7           19.3   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream            55.8           19.8   
340    341  Chinstrap      Dream            43.5           18.1   
341    342  Chinstrap      Dream            49.6           18.2   
342    343  Chinstrap      Dream            50.8           19.0   
343    344  Chinstrap      Dream            50.2           18.7   

     flipper_length_mm  body_mass_g     sex  year  weight  
0                181.0       3750.0    male  2007   3.750  
1                186.0       3800.0  female  2007   3.800  
2                195.0       3250.0  female  2007   3.250  
3                  NaN          NaN     NaN  2007     NaN  
4                193.0       3450.0  female  2007   3.450  
..                 ...          ...     ...   ...     ...  
339              207.0       4000.0    male  2009   4.000  
340              202.0       3400.0  female  2009   3.400  
341              193.0       3775.0    male  2009   3.775  
342              210.0       4100.0    male  2009   4.100  
343              198.0       3775.0  female  2009   3.775  

[344 rows x 10 columns]

In [12]:
birds.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
weight                2
dtype: int64

<h1> WHAT TO DO WITH NULL VALUES </h1>

<p> delete row </p>
<p> fill the empty cells with a value </p>

In [13]:
birds['body_mass_g'].mean()

np.float64(4201.754385964912)

In [14]:
birds['sex'].mode()

0    male
Name: sex, dtype: object

In [15]:
birds['bill_length_mm'] = birds['bill_length_mm'].fillna(birds['bill_length_mm'].mean())

In [16]:
birds.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
weight                2
dtype: int64

In [17]:
birds['body_mass_g'] = birds['body_mass_g'].fillna(birds['body_mass_g'].mean())

In [18]:
birds.isnull().sum()

rowid                 0
species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           0
sex                  11
year                  0
weight                2
dtype: int64

In [19]:
birds['sex'] = birds['sex'].fillna(birds['sex'].mode()[0])

In [20]:
birds.isnull().sum()

rowid                0
species              0
island               0
bill_length_mm       0
bill_depth_mm        2
flipper_length_mm    2
body_mass_g          0
sex                  0
year                 0
weight               2
dtype: int64

In [21]:
birds['bill_depth_mm'] = birds['bill_depth_mm'].fillna(birds['bill_depth_mm'].mean())

In [22]:
birds.isnull().sum()

rowid                0
species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    2
body_mass_g          0
sex                  0
year                 0
weight               2
dtype: int64

In [23]:
birds['flipper_length_mm'] = birds['flipper_length_mm'].fillna(birds['flipper_length_mm'].mean())

In [24]:
birds.isnull().sum()

rowid                0
species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
weight               2
dtype: int64

In [25]:
birds['weight'] = birds['weight'].fillna(birds['weight'].mean())

In [26]:
birds.isnull().sum()

rowid                0
species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
weight               0
dtype: int64

In [27]:
birds.describe()

rowid  bill_length_mm  bill_depth_mm  flipper_length_mm  \
count  344.000000      344.000000     344.000000         344.000000   
mean   172.500000       43.921930      17.151170         200.915205   
std     99.448479        5.443643       1.969027          14.020657   
min      1.000000       32.100000      13.100000         172.000000   
25%     86.750000       39.275000      15.600000         190.000000   
50%    172.500000       44.250000      17.300000         197.000000   
75%    258.250000       48.500000      18.700000         213.000000   
max    344.000000       59.600000      21.500000         231.000000   

       body_mass_g         year      weight  
count   344.000000   344.000000  344.000000  
mean   4201.754386  2008.029070    4.201754  
std     799.613058     0.818356    0.799613  
min    2700.000000  2007.000000    2.700000  
25%    3550.000000  2007.000000    3.550000  
50%    4050.000000  2008.000000    4.050000  
75%    4750.000000  2009.000000    4.750000  
max    6300.000000  2009.000000    6.300000

In [28]:
birds

rowid    species     island  bill_length_mm  bill_depth_mm  \
0        1     Adelie  Torgersen        39.10000       18.70000   
1        2     Adelie  Torgersen        39.50000       17.40000   
2        3     Adelie  Torgersen        40.30000       18.00000   
3        4     Adelie  Torgersen        43.92193       17.15117   
4        5     Adelie  Torgersen        36.70000       19.30000   
..     ...        ...        ...             ...            ...   
339    340  Chinstrap      Dream        55.80000       19.80000   
340    341  Chinstrap      Dream        43.50000       18.10000   
341    342  Chinstrap      Dream        49.60000       18.20000   
342    343  Chinstrap      Dream        50.80000       19.00000   
343    344  Chinstrap      Dream        50.20000       18.70000   

     flipper_length_mm  body_mass_g     sex  year    weight  
0           181.000000  3750.000000    male  2007  3.750000  
1           186.000000  3800.000000  female  2007  3.800000  
2           195.000000  3250.000000  female  2007  3.250000  
3           200.915205  4201.754386    male  2007  4.201754  
4           193.000000  3450.000000  female  2007  3.450000  
..                 ...          ...     ...   ...       ...  
339         207.000000  4000.000000    male  2009  4.000000  
340         202.000000  3400.000000  female  2009  3.400000  
341         193.000000  3775.000000    male  2009  3.775000  
342         210.000000  4100.000000    male  2009  4.100000  
343         198.000000  3775.000000  female  2009  3.775000  

[344 rows x 10 columns]

<h1> Change Data Type</h1>

In [29]:
birds.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   rowid              344 non-null    int64  
 1   species            344 non-null    object 
 2   island             344 non-null    object 
 3   bill_length_mm     344 non-null    float64
 4   bill_depth_mm      344 non-null    float64
 5   flipper_length_mm  344 non-null    float64
 6   body_mass_g        344 non-null    float64
 7   sex                344 non-null    object 
 8   year               344 non-null    int64  
 9   weight             344 non-null    float64
dtypes: float64(5), int64(2), object(3)
memory usage: 27.0+ KB


In [30]:
birds['sex'] = birds['sex'].astype('category')

In [31]:
birds['year'] = birds['year'].astype('category')

In [32]:
birds['species'] = birds['species'].astype('category')

In [33]:
birds['island'] = birds['island'].astype('category')

In [34]:
birds['rowid'] = birds['rowid'].astype('category')

In [35]:
birds.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   rowid              344 non-null    category
 1   species            344 non-null    category
 2   island             344 non-null    category
 3   bill_length_mm     344 non-null    float64 
 4   bill_depth_mm      344 non-null    float64 
 5   flipper_length_mm  344 non-null    float64 
 6   body_mass_g        344 non-null    float64 
 7   sex                344 non-null    category
 8   year               344 non-null    category
 9   weight             344 non-null    float64 
dtypes: category(5), float64(5)
memory usage: 26.9 KB


In [36]:
birds.head(10)

rowid species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0     1  Adelie  Torgersen        39.10000       18.70000         181.000000   
1     2  Adelie  Torgersen        39.50000       17.40000         186.000000   
2     3  Adelie  Torgersen        40.30000       18.00000         195.000000   
3     4  Adelie  Torgersen        43.92193       17.15117         200.915205   
4     5  Adelie  Torgersen        36.70000       19.30000         193.000000   
5     6  Adelie  Torgersen        39.30000       20.60000         190.000000   
6     7  Adelie  Torgersen        38.90000       17.80000         181.000000   
7     8  Adelie  Torgersen        39.20000       19.60000         195.000000   
8     9  Adelie  Torgersen        34.10000       18.10000         193.000000   
9    10  Adelie  Torgersen        42.00000       20.20000         190.000000   

   body_mass_g     sex  year    weight  
0  3750.000000    male  2007  3.750000  
1  3800.000000  female  2007  3.800000  
2  3250.000000  female  2007  3.250000  
3  4201.754386    male  2007  4.201754  
4  3450.000000  female  2007  3.450000  
5  3650.000000    male  2007  3.650000  
6  3625.000000  female  2007  3.625000  
7  4675.000000    male  2007  4.675000  
8  3475.000000    male  2007  3.475000  
9  4250.000000    male  2007  4.250000

In [37]:
birds.tail(5)

rowid    species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
339   340  Chinstrap  Dream            55.8           19.8              207.0   
340   341  Chinstrap  Dream            43.5           18.1              202.0   
341   342  Chinstrap  Dream            49.6           18.2              193.0   
342   343  Chinstrap  Dream            50.8           19.0              210.0   
343   344  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  weight  
339       4000.0    male  2009   4.000  
340       3400.0  female  2009   3.400  
341       3775.0    male  2009   3.775  
342       4100.0    male  2009   4.100  
343       3775.0  female  2009   3.775

In [38]:
birds['species'] == "Chinstrap"

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: species, Length: 344, dtype: bool

In [39]:
birds[birds['species'] == "Chinstrap"].count()

rowid                68
species              68
island               68
bill_length_mm       68
bill_depth_mm        68
flipper_length_mm    68
body_mass_g          68
sex                  68
year                 68
weight               68
dtype: int64

In [40]:
birds[birds['species'] == "Adelie"].count()

rowid                152
species              152
island               152
bill_length_mm       152
bill_depth_mm        152
flipper_length_mm    152
body_mass_g          152
sex                  152
year                 152
weight               152
dtype: int64

In [41]:
birds[birds['species'] == "Gentoo"].count()

rowid                124
species              124
island               124
bill_length_mm       124
bill_depth_mm        124
flipper_length_mm    124
body_mass_g          124
sex                  124
year                 124
weight               124
dtype: int64

<h1>HOW MANY GENTOO SPECIES HAVE WEIGHT MORE THAN 3KG </h1>

In [42]:
birds['species'] == "Gentoo"

0      False
1      False
2      False
3      False
4      False
       ...  
339    False
340    False
341    False
342    False
343    False
Name: species, Length: 344, dtype: bool

In [43]:
birds['weight'] > 3

0      True
1      True
2      True
3      True
4      True
       ... 
339    True
340    True
341    True
342    True
343    True
Name: weight, Length: 344, dtype: bool

<h1> DOES GENTOO LIVE ON ONE ISLAND? </h1>

In [44]:
birds

rowid    species     island  bill_length_mm  bill_depth_mm  \
0       1     Adelie  Torgersen        39.10000       18.70000   
1       2     Adelie  Torgersen        39.50000       17.40000   
2       3     Adelie  Torgersen        40.30000       18.00000   
3       4     Adelie  Torgersen        43.92193       17.15117   
4       5     Adelie  Torgersen        36.70000       19.30000   
..    ...        ...        ...             ...            ...   
339   340  Chinstrap      Dream        55.80000       19.80000   
340   341  Chinstrap      Dream        43.50000       18.10000   
341   342  Chinstrap      Dream        49.60000       18.20000   
342   343  Chinstrap      Dream        50.80000       19.00000   
343   344  Chinstrap      Dream        50.20000       18.70000   

     flipper_length_mm  body_mass_g     sex  year    weight  
0           181.000000  3750.000000    male  2007  3.750000  
1           186.000000  3800.000000  female  2007  3.800000  
2           195.000000  3250.000000  female  2007  3.250000  
3           200.915205  4201.754386    male  2007  4.201754  
4           193.000000  3450.000000  female  2007  3.450000  
..                 ...          ...     ...   ...       ...  
339         207.000000  4000.000000    male  2009  4.000000  
340         202.000000  3400.000000  female  2009  3.400000  
341         193.000000  3775.000000    male  2009  3.775000  
342         210.000000  4100.000000    male  2009  4.100000  
343         198.000000  3775.000000  female  2009  3.775000  

[344 rows x 10 columns]

In [45]:
birds['species'] == "Gentoo"

0      False
1      False
2      False
3      False
4      False
       ...  
339    False
340    False
341    False
342    False
343    False
Name: species, Length: 344, dtype: bool

In [46]:
birds['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

In [47]:
birds['island'] == "Biscoe"

0      False
1      False
2      False
3      False
4      False
       ...  
339    False
340    False
341    False
342    False
343    False
Name: island, Length: 344, dtype: bool

In [48]:
(birds['species'] == "Gentoo") & (birds['island'] == "Biscoe")

0      False
1      False
2      False
3      False
4      False
       ...  
339    False
340    False
341    False
342    False
343    False
Length: 344, dtype: bool

In [49]:
birds[(birds['species'] == "Gentoo") & (birds['island'] == "Biscoe")]

rowid species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152   153  Gentoo  Biscoe        46.10000       13.20000         211.000000   
153   154  Gentoo  Biscoe        50.00000       16.30000         230.000000   
154   155  Gentoo  Biscoe        48.70000       14.10000         210.000000   
155   156  Gentoo  Biscoe        50.00000       15.20000         218.000000   
156   157  Gentoo  Biscoe        47.60000       14.50000         215.000000   
..    ...     ...     ...             ...            ...                ...   
271   272  Gentoo  Biscoe        43.92193       17.15117         200.915205   
272   273  Gentoo  Biscoe        46.80000       14.30000         215.000000   
273   274  Gentoo  Biscoe        50.40000       15.70000         222.000000   
274   275  Gentoo  Biscoe        45.20000       14.80000         212.000000   
275   276  Gentoo  Biscoe        49.90000       16.10000         213.000000   

     body_mass_g     sex  year    weight  
152  4500.000000  female  2007  4.500000  
153  5700.000000    male  2007  5.700000  
154  4450.000000  female  2007  4.450000  
155  5700.000000    male  2007  5.700000  
156  5400.000000    male  2007  5.400000  
..           ...     ...   ...       ...  
271  4201.754386    male  2009  4.201754  
272  4850.000000  female  2009  4.850000  
273  5750.000000    male  2009  5.750000  
274  5200.000000  female  2009  5.200000  
275  5400.000000    male  2009  5.400000  

[124 rows x 10 columns]

In [50]:
birds[(birds['species'] == "Gentoo") & (birds['weight'] > 5 )].count()

rowid                61
species              61
island               61
bill_length_mm       61
bill_depth_mm        61
flipper_length_mm    61
body_mass_g          61
sex                  61
year                 61
weight               61
dtype: int64

<h1>ACTIVITY 2</h1>

In [51]:
birds['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [52]:
birds['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

<H1>GENTOO</H1>

In [53]:
gentoobiscoe = birds[(birds['species'] == "Gentoo") & (birds['island'] == "Biscoe" )]
gentoobiscoe

rowid species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152   153  Gentoo  Biscoe        46.10000       13.20000         211.000000   
153   154  Gentoo  Biscoe        50.00000       16.30000         230.000000   
154   155  Gentoo  Biscoe        48.70000       14.10000         210.000000   
155   156  Gentoo  Biscoe        50.00000       15.20000         218.000000   
156   157  Gentoo  Biscoe        47.60000       14.50000         215.000000   
..    ...     ...     ...             ...            ...                ...   
271   272  Gentoo  Biscoe        43.92193       17.15117         200.915205   
272   273  Gentoo  Biscoe        46.80000       14.30000         215.000000   
273   274  Gentoo  Biscoe        50.40000       15.70000         222.000000   
274   275  Gentoo  Biscoe        45.20000       14.80000         212.000000   
275   276  Gentoo  Biscoe        49.90000       16.10000         213.000000   

     body_mass_g     sex  year    weight  
152  4500.000000  female  2007  4.500000  
153  5700.000000    male  2007  5.700000  
154  4450.000000  female  2007  4.450000  
155  5700.000000    male  2007  5.700000  
156  5400.000000    male  2007  5.400000  
..           ...     ...   ...       ...  
271  4201.754386    male  2009  4.201754  
272  4850.000000  female  2009  4.850000  
273  5750.000000    male  2009  5.750000  
274  5200.000000  female  2009  5.200000  
275  5400.000000    male  2009  5.400000  

[124 rows x 10 columns]

In [54]:
gentoodream = birds[(birds['species'] == "Gentoo") & (birds['island'] == "Dream" )]
gentoodream.info()

<class 'pandas.core.frame.DataFrame'>
Index: 0 entries
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   rowid              0 non-null      category
 1   species            0 non-null      category
 2   island             0 non-null      category
 3   bill_length_mm     0 non-null      float64 
 4   bill_depth_mm      0 non-null      float64 
 5   flipper_length_mm  0 non-null      float64 
 6   body_mass_g        0 non-null      float64 
 7   sex                0 non-null      category
 8   year               0 non-null      category
 9   weight             0 non-null      float64 
dtypes: category(5), float64(5)
memory usage: 11.3 KB


In [55]:
gentootorgersen = birds[(birds['species'] == "Gentoo") & (birds['island'] == "Torgersen" )]
gentootorgersen.info()

<class 'pandas.core.frame.DataFrame'>
Index: 0 entries
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   rowid              0 non-null      category
 1   species            0 non-null      category
 2   island             0 non-null      category
 3   bill_length_mm     0 non-null      float64 
 4   bill_depth_mm      0 non-null      float64 
 5   flipper_length_mm  0 non-null      float64 
 6   body_mass_g        0 non-null      float64 
 7   sex                0 non-null      category
 8   year               0 non-null      category
 9   weight             0 non-null      float64 
dtypes: category(5), float64(5)
memory usage: 11.3 KB


<H3>1. GENTOO LIVE ON ONE ISLAND (BISCOE)</H3>

<h1>ADELIE</h1>

In [56]:
adeliebiscoe = birds[(birds['species'] == "Adelie") & (birds['island'] == "Biscoe" )]
adeliebiscoe.info()

<class 'pandas.core.frame.DataFrame'>
Index: 44 entries, 20 to 115
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   rowid              44 non-null     category
 1   species            44 non-null     category
 2   island             44 non-null     category
 3   bill_length_mm     44 non-null     float64 
 4   bill_depth_mm      44 non-null     float64 
 5   flipper_length_mm  44 non-null     float64 
 6   body_mass_g        44 non-null     float64 
 7   sex                44 non-null     category
 8   year               44 non-null     category
 9   weight             44 non-null     float64 
dtypes: category(5), float64(5)
memory usage: 13.6 KB


In [57]:
adeliedream = birds[(birds['species'] == "Adelie") & (birds['island'] == "Dream" )]
adeliedream.info()

<class 'pandas.core.frame.DataFrame'>
Index: 56 entries, 30 to 151
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   rowid              56 non-null     category
 1   species            56 non-null     category
 2   island             56 non-null     category
 3   bill_length_mm     56 non-null     float64 
 4   bill_depth_mm      56 non-null     float64 
 5   flipper_length_mm  56 non-null     float64 
 6   body_mass_g        56 non-null     float64 
 7   sex                56 non-null     category
 8   year               56 non-null     category
 9   weight             56 non-null     float64 
dtypes: category(5), float64(5)
memory usage: 14.2 KB


In [73]:
adelietorgersen = birds[(birds['species'] == "Adelie") & (birds['island'] == "Torgersen" )]
adelietorgersen.info()

<class 'pandas.core.frame.DataFrame'>
Index: 52 entries, 0 to 131
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   rowid              52 non-null     category
 1   species            52 non-null     category
 2   island             52 non-null     category
 3   bill_length_mm     52 non-null     float64 
 4   bill_depth_mm      52 non-null     float64 
 5   flipper_length_mm  52 non-null     float64 
 6   body_mass_g        52 non-null     float64 
 7   sex                52 non-null     category
 8   year               52 non-null     category
 9   weight             52 non-null     float64 
dtypes: category(5), float64(5)
memory usage: 14.0 KB


<H3>2. ADELIE LIVE ON ALL 3 ISLANDS (BISCOE, TORGERSEN, DREAM)</H3>

<H1>CHINSTRAP</H1>

In [74]:
chinstrapbiscoe = birds[(birds['species'] == "Chinstrap") & (birds['island'] == "Biscoe" )]
chinstrapbiscoe.info()

<class 'pandas.core.frame.DataFrame'>
Index: 0 entries
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   rowid              0 non-null      category
 1   species            0 non-null      category
 2   island             0 non-null      category
 3   bill_length_mm     0 non-null      float64 
 4   bill_depth_mm      0 non-null      float64 
 5   flipper_length_mm  0 non-null      float64 
 6   body_mass_g        0 non-null      float64 
 7   sex                0 non-null      category
 8   year               0 non-null      category
 9   weight             0 non-null      float64 
dtypes: category(5), float64(5)
memory usage: 11.3 KB


In [75]:
chinstrapdream = birds[(birds['species'] == "Chinstrap") & (birds['island'] == "Dream" )]
chinstrapdream.info()

<class 'pandas.core.frame.DataFrame'>
Index: 68 entries, 276 to 343
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   rowid              68 non-null     category
 1   species            68 non-null     category
 2   island             68 non-null     category
 3   bill_length_mm     68 non-null     float64 
 4   bill_depth_mm      68 non-null     float64 
 5   flipper_length_mm  68 non-null     float64 
 6   body_mass_g        68 non-null     float64 
 7   sex                68 non-null     category
 8   year               68 non-null     category
 9   weight             68 non-null     float64 
dtypes: category(5), float64(5)
memory usage: 14.9 KB


In [76]:
chinstraptorgersen = birds[(birds['species'] == "Chinstrap") & (birds['island'] == "Torgersen" )]
chinstraptorgersen.info()

<class 'pandas.core.frame.DataFrame'>
Index: 0 entries
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   rowid              0 non-null      category
 1   species            0 non-null      category
 2   island             0 non-null      category
 3   bill_length_mm     0 non-null      float64 
 4   bill_depth_mm      0 non-null      float64 
 5   flipper_length_mm  0 non-null      float64 
 6   body_mass_g        0 non-null      float64 
 7   sex                0 non-null      category
 8   year               0 non-null      category
 9   weight             0 non-null      float64 
dtypes: category(5), float64(5)
memory usage: 11.3 KB


<H3>3. CHINSTRAP LIVE ON ONE ISLAND (DREAM)</H3>

In [77]:
birds['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [78]:
birds['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

<h1>WEIGHT INSIGHTS</h1>

In [79]:
adeliebiscoe_w = np.round(adeliebiscoe['weight'].mean(), 1)
adeliebiscoe_w

np.float64(3.7)

In [80]:
adelietorgersen_w = np.round(adelietorgersen['weight'].mean(), 1)
adelietorgersen_w

np.float64(3.7)

In [81]:
adeliedream_w = np.round(adeliedream['weight'].mean(), 1)
adeliedream_w

np.float64(3.7)

<h3>4. THERE IS NO IMPACT IN THE WEIGHT OF ADELIE ON EACH ISLAND</h3>

In [82]:
gentoobiscoe_w = np.round(gentoobiscoe['weight'].mean(), 1)
gentoobiscoe_w

np.float64(5.1)

<h3>5. GENTOO'S AVERAGE WEIGHT ON BISCOE (5.1)</h3>

In [68]:
chinstrapdream_w = np.round(chinstrapdream['weight'].mean(), 1)
chinstrapdream_w

np.float64(3.7)

<h3>6. CHINSTRAP'S AVERAGE WEIGHT ON DREAM (3.7)</h3>

<H3>7. HEAVIEST SPECIE IS GENTOO WITH AVERAGE WEIGHT OF 5kg</H3>